core/vm/interface.go

```diff
@@ -26,58 +26,88 @@ import (
+	// For non-existent addresses or non-existent keys it returns empty hash (common.Hash{})
+	GetState(addr common.Address, key common.Hash) common.Hash
+	// SetState sets a value for the given (address, key) pair (aka slot)
+	// If a value already exist at the given pair, it will be replaced.
+	SetState(addr common.Address, key common.Hash, value common.Hash)
+	// GetCommittedState returns the value for the given (address, key) pair (aka slot) in the transient state
+	// Check EIP-1153 for more info about transient state
 	GetTransientState(addr common.Address, key common.Hash) common.Hash
+	// SetTransientState sets a value for the given (address, key) pair (aka slot) in the transient state
+	// Check EIP-1153 for more info about transient state
 	SetTransientState(addr common.Address, key, value common.Hash)
-
+	// SelfDestruct flags an address for destruction at the end of a transaction
 	SelfDestruct(common.Address)
+	// HasSelfDestructed returns true if address has been flaged for destruction (during this transaction)
 	HasSelfDestructed(common.Address) bool
-
+	// Selfdestruct6780 implements the logic of self destruct according to the EIP-6700
 	Selfdestruct6780(common.Address)
-
 	// Exist reports whether the given account exists in state.
-	// Notably this should also return true for self-destructed accounts.
+	// Notably this returns true for self-destructed accounts.
 	Exist(common.Address) bool
 	// Empty returns whether the given account is empty. Empty
-	// is defined according to EIP161 (balance = nonce = code = 0).
+	// is defined according to EIP-161 (balance = nonce = code = 0).
 	Empty(common.Address) bool
-
+	// AddressInAccessList checks if an address is in the access list (see EIP-2930).
 	AddressInAccessList(addr common.Address) bool
+	// SlotInAccessList checks if an slot is in the access list (see EIP-2930).
 	SlotInAccessList(addr common.Address, slot common.Hash) (addressOk bool, slotOk bool)
 	// AddAddressToAccessList adds the given address to the access list. This operation is safe to perform
-	// even if the feature/fork is not active yet
+	// even if the feature/fork is not active yet (see EIP-2930).
 	AddAddressToAccessList(addr common.Address)
 	// AddSlotToAccessList adds the given (address,slot) to the access list. This operation is safe to perform
-	// even if the feature/fork is not active yet
+	// even if the feature/fork is not active yet (see EIP-2930).
 	AddSlotToAccessList(addr common.Address, slot common.Hash)
+	// Prepare prepares the storage for the next transaction
+	// it resets the access list and transient storage,
+	// it also adds addresses and slots to the list based on the rules of the network.
 	Prepare(rules params.Rules, sender, coinbase common.Address, dest *common.Address, precompiles []common.Address, txAccesses types.AccessList)
-
+	// RevertToSnapshot reverts the state to an specific snapshot
```

Comment:
Why not reuse the existing doc in `statedb.go`
```suggestion
	// RevertToSnapshot reverts all state changes made since the given revision.
```